pkg/controller/podgc/gc_controller.go

```diff
@@ -308,7 +316,30 @@ func (o byCreationTimestamp) Less(i, j int) bool {
 	return o[i].CreationTimestamp.Before(&o[j].CreationTimestamp)
 }
 
-func (gcc *PodGCController) deletePod(ctx context.Context, namespace, name string) error {
-	klog.InfoS("PodGC is force deleting Pod", "pod", klog.KRef(namespace, name))
-	return gcc.kubeClient.CoreV1().Pods(namespace).Delete(ctx, name, *metav1.NewDeleteOptions(0))
+func (gcc *PodGCController) markFailedAndDeletePod(ctx context.Context, pod *v1.Pod) error {
+	return gcc.markFailedAndDeletePodWithCondition(ctx, pod, nil)
+}
+
+func (gcc *PodGCController) markFailedAndDeletePodWithCondition(ctx context.Context, pod *v1.Pod, condition *v1.PodCondition) error {
+	klog.InfoS("PodGC is force deleting Pod", "pod", klog.KRef(pod.Namespace, pod.Name))
+	if utilfeature.DefaultFeatureGate.Enabled(features.PodDisruptionConditions) {
+		newStatus := pod.Status.DeepCopy()
+		updated := false
+		if condition != nil {
+			updated = apipod.UpdatePodCondition(newStatus, condition)
+		}
+		// Mark the pod as failed - this is especially important in case the pod
+		// is orphaned, in which case the pod would remain in the Running phase
+		// forever as there is no kubelet running to change the phase.
+		if pod.Status.Phase != v1.PodSucceeded && pod.Status.Phase != v1.PodFailed {
+			newStatus.Phase = v1.PodFailed
+			updated = true
+		}
+		if updated {
+			if _, _, _, err := utilpod.PatchPodStatus(ctx, gcc.kubeClient, pod.Namespace, pod.Name, pod.UID, pod.Status, *newStatus); err != nil {
+				return err
```

Comment:
should you try to delete the pod always?

Reply:
No, we need the information, we should let the controller retry.

Reply:
Good point, @alculquicondor  @soltysh ? I suggest to maybe re-evaluate the decision for Beta. Anyway for Beta we are planning to commonize the the code for appending the pod conditions.